### PR TITLE
[DX-1002] Improved Ignore Authentication

### PR DESCRIPTION
**Pull request by andyo-tyk**
### Preview Link
https://deploy-preview-4031--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/middleware/ignore-middleware/

### Description
Improved documentation of Ignore Auth middleware

### CODE REVIEW

File: tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-middleware.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+## When to use the ignore authentication middleware
+#### Health and liveness endpoints
+This plugin can be very useful if you have an endpoint (such as a ping or health check) that you don’t need to secure.
+
+## How ignore authentication works
+When the Ignore Authentication middleware is configured for a specific endpoint, it instructs the gateway to bypass the client authentication process for requests made to that endpoint. If other (non-authentication) middleware are configured for the endpoint, they will still execute on the request.
+
+It is important to exercise caution when using the Ignore Authentication middleware, as it effectively disables Tyk's security features for the ignored paths. Only endpoints that are designed to be public or have independent security mechanisms should be configured to bypass authentication in this way. It is particularly important to be careful when combining Ignore Authentication with response transformations not to inadvertently expose sensitive data or rely on authentication or session data that is not present.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-middleware.md]:
```suggestion
It is important to exercise caution when using the Ignore Authentication middleware, as it effectively disables Tyk's security features for the ignored paths. Only endpoints that are designed to be public or have independent security mechanisms should be configured to bypass authentication in this way. When combining Ignore Authentication with response transformations be careful not to inadvertently expose sensitive data or rely on authentication or session data that is not present.
```

File: tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-classic.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+---
+title: Using the Ignore Authentication middleware with Tyk Classic APIs
+date: 2024-01-24
+description: "Using the Ignore Authentication middleware with Tyk Classic APIs"
+tags: ["ignore authentication", "ignore", "ignore auth", "authentication", "middleware", "per-endpoint", "Tyk Classic"]
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-classic.md]:
```suggestion
tags: ["ignore authentication", "ignore", "ignore auth", "authentication", "middleware", "per-endpoint", "Tyk Classic", "Tyk Classic APIs"]
```

File: tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-classic.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+---
+
+The [Ignore Authentication]({{< ref "product-stack/tyk-gateway/middleware/ignore-middleware" >}}) middleware instructs Tyk Gateway to skip the authentication step for calls to an endpoint, even if authentication is enabled for the API.
+
+When working with Tyk Classic APIs the middleware is configured in the Tyk Classic API Definition either manually within the `.json` file or from the API Designer in the Tyk Dashboard.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-classic.md]:
```suggestion
When working with Tyk Classic APIs the middleware is configured in the Tyk Classic API Definition, either manually within the `.json` file or from the API Designer in the Tyk Dashboard.
```

File: tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-classic.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                    }          
+                }
+            }
+        ]
+    }
+}
+```
+
+In this example the ignore authentication middleware has been configured for HTTP `GET` requests to the `/status/200` endpoint. Any such calls will skip the authentication step in the Tyk Gateway's processing chain.
+ - the allow list has been configured to be case sensitive, so calls to `GET /Status/200` will not skip authentication
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-classic.md]:
```suggestion
 - The allow list has been configured to be case sensitive, so calls to `GET /Status/200` will not skip authentication
```

File: tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-oas.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
+---
+title: Using the Ignore Authentication middleware with Tyk OAS APIs
+date: 2024-01-24
+description: "Using the Ignore Authentication middleware with Tyk OAS APIs"
+tags: ["ignore authentication", "ignore", "ignore auth", "authentication", "middleware", "per-endpoint", "Tyk OAS"]
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-oas.md]:
```suggestion
tags: ["ignore authentication", "ignore", "ignore auth", "authentication", "middleware", "per-endpoint", "Tyk OAS", "Tyk OAS APIs"]
```

File: tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-oas.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+                    }                
+                }
+            }
+        }
+    }
+}
+```
+
+In this example the ignore authentication middleware has been configured for HTTP `GET` requests to the `/status/200` endpoint. Any such calls will skip the authentication step in the Tyk Gateway's processing chain.
+ - the allow list has been configured to be case sensitive, so calls to `GET /Status/200` will not skip authentication
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-oas.md]:
```suggestion
 - The allow list has been configured to be case sensitive, so calls to `GET /Status/200` will not skip authentication
```

File: tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-oas.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+The configuration above is a complete and valid Tyk OAS API Definition that you can import into Tyk to try out the API-level response header transform.
+
+## Configuring the middleware in the API Designer
+Adding Ignore Authentication to your API endpoints is easy when using the API Designer in the Tyk Dashboard, simply follow the steps taken in this short video:
+
+ < placeholder for video >
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-oas.md]: Remember to add placeholder video

**Review by dcs3spp — CHANGES_REQUESTED** [review]
PR LGTM Andy. I've added the following minor suggestions below, then once approved and peer reviewed we can mark PR as ready for release

- Minor grammar suggestions (Done)
- Remove H2 ## overview headings (Done)
- _Remember to add placeholder video_

Will mark as draft until placeholder video available

File: tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-classic.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
+---
+title: Using the Ignore Authentication middleware with Tyk Classic APIs
+date: 2024-01-24
+description: "Using the Ignore Authentication middleware with Tyk Classic APIs"
+tags: ["ignore authentication", "ignore", "ignore auth", "authentication", "middleware", "per-endpoint", "Tyk Classic", "Tyk Classic APIs"]
+---
+
+The [Ignore Authentication]({{< ref "product-stack/tyk-gateway/middleware/ignore-middleware" >}}) middleware instructs Tyk Gateway to skip the authentication step for calls to an endpoint, even if authentication is enabled for the API.
+
+When working with Tyk Classic APIs the middleware is configured in the Tyk Classic API Definition, either manually within the `.json` file or from the API Designer in the Tyk Dashboard.
```
Comment (andyo-tyk) [inline tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-classic.md]:
```suggestion
When working with Tyk Classic APIs the middleware is configured in the Tyk Classic API Definition. You can do this via the Tyk Dashboard API or in the API Designer.
```

File: tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-oas.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
+---
+title: Using the Ignore Authentication middleware with Tyk OAS APIs
+date: 2024-01-24
+description: "Using the Ignore Authentication middleware with Tyk OAS APIs"
+tags: ["ignore authentication", "ignore", "ignore auth", "authentication", "middleware", "per-endpoint", "Tyk OAS", "Tyk OAS APIs"]
+---
+
+The [Ignore Authentication]({{< ref "product-stack/tyk-gateway/middleware/ignore-middleware" >}}) middleware instructs Tyk Gateway to skip the authentication step for calls to an endpoint, even if authentication is enabled for the API.
+
+When working with Tyk OAS APIs the middleware is configured in the [Tyk OAS API Definition]({{< ref "tyk-apis/tyk-gateway-api/oas/x-tyk-oas-doc#operation" >}}) either manually within the `.json` file or from the API Designer in the Tyk Dashboard.
```
Comment (andyo-tyk) [inline tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-oas.md]:
```suggestion
When working with Tyk OAS APIs the middleware is configured in the [Tyk OAS API Definition]({{< ref "tyk-apis/tyk-gateway-api/oas/x-tyk-oas-doc#operation" >}}). You can do this via the Tyk Dashboard API or in the API Designer.
```

**Review by andyo-tyk — COMMENTED** [review]
Minor improvements

**Comment by dcs3spp** [comment]
@andyo-tyk To facilitate reviewing the PR queue we are closing some PRs that are in draft status, e.g. awaiting docker images, videos, images etc. You can still update the PR and then open it again when ready for review.

cc: @letzya

**Comment by titpetric** [comment]
Reopening for peer review

File: tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-middleware.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+title: Ignore Authentication middleware
+date: 2024-01-24
+description: "Detail of the Ignore Authentication middleware"
+tags: ["ignore authentication", "ignore", "ignore auth", "authentication", "middleware", "per-endpoint"]
+---
+
+The Ignore Authentication middleware instructs Tyk Gateway to skip the authentication step for calls to an endpoint, even if authentication is enabled for the API.
+
+## When to use the ignore authentication middleware
+#### Health and liveness endpoints
```
Comment (titpetric) [inline tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-middleware.md:11]: I expected a list of possible use cases here, but only one example is listed.

File: tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-middleware.md [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+This plugin can be very useful if you have an endpoint (such as a ping or health check) that you don’t need to secure.
+
+## How ignore authentication works
+When the Ignore Authentication middleware is configured for a specific endpoint, it instructs the gateway to bypass the client authentication process for requests made to that endpoint. If other (non-authentication) middleware are configured for the endpoint, they will still execute on the request.
+
+It is important to exercise caution when using the Ignore Authentication middleware, as it effectively disables Tyk's security features for the ignored paths. Only endpoints that are designed to be public or have independent security mechanisms should be configured to bypass authentication in this way. When combining Ignore Authentication with response transformations be careful not to inadvertently expose sensitive data or rely on authentication or session data that is not present.
+
+#### Case sensitivity
```
Comment (titpetric) [inline tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-middleware.md]: The headings in this doc are a bit confusing

- h2 When to use [...]
    - h4 Health and liveliness endpoints
- h2 How ignore authentication works
    - h4 Case sensitivity

This produces a table of contents that doesn't seem balanced with information, suggesting to reorder/redefine the sections in order to make a sensible TOC

**Review by titpetric — APPROVED** [review]
LGTM from a technical perspective, left comments on heading/order in the overview doc.

File: tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-middleware.md (hunk 10 above)
Comment (andyo-tyk) [reply tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-middleware.md:11]: I'm adding these retrospectively to existing (undocumented) features - if you can suggest others then please do. As in - I'd love to have more, we can always add them later. 🙏

File: tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-middleware.md (hunk 11 above)
Comment (andyo-tyk) [reply tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-middleware.md]: This is consistent across the various PRs. The reason I've not used every level of heading is that the style-sheet used for Tyk docs does not clearly show a difference between different levels so it's actually harder to read, even if the ToC becomes clearer... and the pages are intended to be short such that ToC is not so important.

File: tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-oas.md (hunk 7 above)
Comment (andyo-tyk) [reply tyk-docs/content/product-stack/tyk-gateway/middleware/ignore-tyk-oas.md]: I have added screenshots for now

**Comment by andyo-tyk** [comment]
Closing, as this has been merged in https://github.com/TykTechnologies/tyk-docs/pull/4318

**Pull request closed without merging**